### PR TITLE
Add GitVersion

### PR DESCRIPTION
**Pull request by david-driscoll**
* Adds GitVersion
* Removes `RunTool`
  * Replaced with calls to `DotNetCoreX()`
* Modified how test assets are created to utilize `DoesForEach`

### CODE REVIEW

**Comment by david-driscoll** [comment]
I'm actually shocked that this worked on the first try.  Only the mac build is currently failing (damn spastic thing)

**Comment by david-driscoll** [comment]
I was thinking Continuous deployment, because what the goal here is to push packages to a azure blob storage for testing purposes.

File: build.cake [hunk 1]
```diff
@@ -353,75 +362,98 @@ Task("Restore")
...
-    // Restore and build Windows-only test assets
-    if (platform.IsWindows)
-    {
-        foreach (var project in buildPlan.WindowsOnlyTestAssets)
+        DotNetCoreRestore(new DotNetCoreRestoreSettings()
         {
-            Information("Restoring and building: {0} (windows-only)...", project);
+            ToolPath = env.DotNetCommand,
+            WorkingDirectory = folder
+            });
```
Comment (DustinCampbell) [inline build.cake]: nit: extra indent

File: build.cake [hunk 2]
```diff
@@ -353,75 +362,98 @@ Task("Restore")
...
-        // Restore and build legacy test assets with legacy .NET Core SDK
-        foreach (var project in buildPlan.LegacyTestAssets)
-        {
-            Information("Restoring and building project.json: {0}...", project);
+        // RunTool(env.LegacyDotNetCommand, "restore", folder)
+        //     .ExceptionOnError($"Failed to restore '{folder}'.");
 
-            var folder = CombinePaths(env.Folders.TestAssets, "test-projects", project);
+        // RunTool(env.LegacyDotNetCommand, $"build", folder)
+        //     .ExceptionOnError($"Failed to build '{folder}'.");
```
Comment (DustinCampbell) [inline build.cake]: Did you mean to leave these here commented?

File: build.cake [hunk 3]
```diff
@@ -88,10 +88,19 @@ Task("Cleanup")
     DirectoryHelper.Create(env.Folders.ArtifactsScripts);
 });
 
+Task("GitVersion")
+    // .WithCriteria(!BuildSystem.IsLocalBuild)
```
Comment (rchande) [inline build.cake]: Should this be uncommented?

Comment (david-driscoll) [reply build.cake]: yeah, probably, will fix.

File: build.cake (hunk 2 above)
Comment (david-driscoll) [reply build.cake]: Nope, they were for reference to make sure I had made the correct calls.  Fixing.

**Comment by david-driscoll** [comment]
Right now this change seems to extend the mac builds so long that they cannot complete in a timely manner.  I'll look at the reducing the logging verbosity to see if that helps.

**Comment by david-driscoll** [comment]
@mholo65 running into a strange issue with a few cake tests failing for some reason.

https://travis-ci.org/OmniSharp/omnisharp-roslyn/jobs/318197803

Errors I'm seeing

```
 OmniSharp.Cake.Tests.UpdateBufferFacts.ShouldSupportIncrementalChanges [FAIL]

      System.InvalidOperationException : Service not initialized.

      Stack Trace:

          at OmniSharp.Cake.Services.CakeScriptService.Generate (Cake.Scripting.Abstractions.Models.FileChange fileChange) [0x0000f] in <04adc11cce8d4bb3bbdfe7fbfc5864d3>:0 

          at OmniSharp.Cake.Services.RequestHandlers.Buffer.UpdateBufferHandler+<Handle>d__3.MoveNext () [0x0016f] in <04adc11cce8d4bb3bbdfe7fbfc5864d3>:0 

        --- End of stack trace from previous location where exception was thrown ---

          at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <c9f8153c41de4f8cbafd0e32f9bf6b28>:0 

          at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <c9f8153c41de4f8cbafd0e32f9bf6b28>:0 

          at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <c9f8153c41de4f8cbafd0e32f9bf6b28>:0 

          at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <c9f8153c41de4f8cbafd0e32f9bf6b28>:0 

          at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <c9f8153c41de4f8cbafd0e32f9bf6b28>:0 

          at OmniSharp.Cake.Tests.UpdateBufferFacts+<ShouldSupportIncrementalChanges>d__5.MoveNext () [0x001a3] in <bb1b40d88bd24f2bac2223612db1bb4d>:0 

        --- End of stack trace from previous location where exception was thrown ---

          at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <c9f8153c41de4f8cbafd0e32f9bf6b28>:0 

          at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <c9f8153c41de4f8cbafd0e32f9bf6b28>:0 

          at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <c9f8153c41de4f8cbafd0e32f9bf6b28>:0 

          at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <c9f8153c41de4f8cbafd0e32f9bf6b28>:0 

          at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <c9f8153c41de4f8cbafd0e32f9bf6b28>:0 

        --- End of stack trace from previous location where exception was thrown ---

          at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <c9f8153c41de4f8cbafd0e32f9bf6b28>:0 

          at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <c9f8153c41de4f8cbafd0e32f9bf6b28>:0 

          at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <c9f8153c41de4f8cbafd0e32f9bf6b28>:0 

          at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <c9f8153c41de4f8cbafd0e32f9bf6b28>:0 

          at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <c9f8153c41de4f8cbafd0e32f9bf6b28>:0 

        --- End of stack trace from previous location where exception was thrown ---

          at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <c9f8153c41de4f8cbafd0e32f9bf6b28>:0 

          at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <c9f8153c41de4f8cbafd0e32f9bf6b28>:0 

          at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <c9f8153c41de4f8cbafd0e32f9bf6b28>:0 

          at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <c9f8153c41de4f8cbafd0e32f9bf6b28>:0 

          at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <c9f8153c41de4f8cbafd0e32f9bf6b28>:0
```

I don't quiet understand how things could be broken, but that appears to the be issue with mac builds only though.

**Comment by david-driscoll** [comment]
@mholo65 nevermind I think I found the issue, it was a `ID10T` error on my part.

**Comment by mholo65** [comment]
@david-driscoll, the error is referring to `cake.bakery` not installed. It should be installed during test-asset preparation. Are all tests failing?

**Comment by david-driscoll** [comment]
yay it worked!

**Pull request merged**